packages/drivers/iframe-driver/src/innerDocumentService.ts

```diff
@@ -22,16 +25,32 @@ export class InnerDocumentService implements IDocumentService {
     /**
      * Create a new InnerDocumentService
      */
-    public static async create(proxyObject: ICombinedDriver): Promise<InnerDocumentService> {
-        return new InnerDocumentService(proxyObject, proxyObject.clientId);
+    public static async create(
+        proxyObject: ICombinedDriver,
+        logger?: ITelemetryBaseLogger,
+    ): Promise<InnerDocumentService> {
+        return new InnerDocumentService(proxyObject, proxyObject.clientId, logger);
     }
 
+    private readonly logger: ITelemetryBaseLogger;
+
     private constructor(
         private readonly outerProxy: ICombinedDriver,
-        public clientId: string) { }
+        public clientId: string,
+        logger?: ITelemetryBaseLogger)
+    {
+        // outerProxy provides a logger as well, but we can create and use our own here
+        // for different functionality
+        this.logger = logger
+            ? ChildLogger.create(logger, "InnerIFrameDriver")
+            : new IFrameDebugLogger("InnerIFrameDriver");
```

Comment:
i don't think we should create a logger by default. we should take an optional logger, and produce a combined logger with that and the outproxy logger

Reply:
MultiSinkLogger is the combined logger

---
In reply to: [513623391](https://github.com/microsoft/FluidFramework/pull/4106#discussion_r513623391) [](ancestors = 513623391)